cs/src/Contracts/TunnelOptions.cs

```diff
@@ -52,6 +56,7 @@ public class TunnelOptions
         /// The option is ignored if IsOriginHeaderUnchanged is true.
         /// </summary>
         [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
+        [StringLength(HeaderMaxLength)]
```

Comment:
The origin is {scheme:}//{host}:{port}
This also includes a scheme, so the max length is 255 + 1 + 5 + 8 ,
where max DNS name length is 255, + 1 for ':', + 5 for '65535', max port length, and +8 for 'https://', and  +1 for trailing '/'

Reply:
I think there's no need to be so exact. My intention for this validation is to prevent unreasonably large inputs, so I'm just going to use 300 for both headers for now. You can change that or add regex validation for these later.